### PR TITLE
ENH: Append

### PR DESCRIPTION
**Pull request by fraimondo**
This will add support for appending titles to existing files and updating titles values if they already exist.

### CODE REVIEW

**Comment by dengemann** [comment]
cc @Eric89GXL

**Comment by dengemann** [comment]
Everybody is happy here? Travis is happy. We're happy.

File: h5io/_h5io.py [hunk 1]
```diff
@@ -66,17 +66,26 @@ def write_hdf5(fname, data, overwrite=False, compression=4,
     title : str
         The top-level directory name to use. Typically it is useful to make
         this your package name, e.g. ``'mnepython'``.
+    update_title : bool
+        If True, override value for title (if title exists). If False and title
+        exists and error will be raised.
```
Comment (Eric89GXL) [inline h5io/_h5io.py]: maybe it should just be `overwrite=True | False | 'update'`?

File: h5io/_h5io.py [hunk 2]
```diff
@@ -66,17 +66,26 @@ def write_hdf5(fname, data, overwrite=False, compression=4,
...
     comp_kw = dict()
     if compression > 0:
         comp_kw = dict(compression='gzip', compression_opts=compression)
-    with h5py.File(fname, mode='w') as fid:
+    with h5py.File(fname, mode=mode) as fid:
+        if title in fid:
+            if not update_title:
+                raise IOError('title %s already exists, use update_title=True'
+                              'to override')
+            else:
```
Comment (Eric89GXL) [inline h5io/_h5io.py]: this doesn't actually need to be in an `else` block since the one above raises an error

File: h5io/tests/test_io.py [hunk 3]
```diff
@@ -38,6 +38,27 @@ def test_hdf5():
                   overwrite=True)
     assert_raises(TypeError, write_hdf5, test_file, object, overwrite=True)
 
+    write_hdf5(test_file, 1, title='first')
+    write_hdf5(test_file, 2, title='second')
```
Comment (Eric89GXL) [inline h5io/tests/test_io.py]: Before this PR, this line would have completely overwritten `test_file`, right? If so, this is a backward incompatible change.

Comment (Eric89GXL) [inline h5io/tests/test_io.py]: ...please restore it to its old behavior, and add a test that it does behave that way.

Comment (fraimondo) [inline h5io/tests/test_io.py]: No, you have to explicit overwrite = True, the default would have raised an error. I will go back to the default raise error behaviour and make overwrite = 'update' to append (or replace value if title already exists)

**Comment by dengemann** [comment]
ping can we merge?

File: h5io/_h5io.py [hunk 4]
```diff
@@ -68,15 +69,21 @@ def write_hdf5(fname, data, overwrite=False, compression=4,
...
     h5py = _check_h5py()
-    if op.isfile(fname) and not overwrite:
-        raise IOError('file "%s" exists, use overwrite=True to overwrite'
-                      % fname)
+    mode = 'w'
+    if op.isfile(fname):
+        if overwrite is False:
+            raise IOError('file "%s" exists, use overwrite=True to overwrite'
+                          % fname)
+        if overwrite is 'update':
```
Comment (Eric89GXL) [inline h5io/_h5io.py]: don't use `is` for str comparison, use `if overwriite == 'update'`. Also add check for overwrite being `update` if of type `string_types`. So something like this:
```
if op.isfile(fname):
    if isinstance(overwrite, string_types):
        if overwrite != 'update':
            raise ValueError('overwrite must be "update" or a bool')
        mode = 'a'
    elif not overwrite:
        raise IOError(...)
```

**Comment by Eric89GXL** [comment]
Otherwise LGTM

**Comment by fraimondo** [comment]
Done!

All test pass, ready to merge.

**Comment by Eric89GXL** [comment]
Thanks @fraimondo!

**Pull request merged**